src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -1963,5 +1963,62 @@ public TSource FirstOrDefault(Func<TSource, bool> predicate, params TSource[] de
             Assert.Equal("String", typeInfo.Type.Name);
             Assert.NotEmpty(typeInfo.Type.GetMembers("Replace"));
         }
+
+        [Fact]
+        [WorkItem(557, "https://github.com/dotnet/roslyn/issues/557")]
+        public void TestLambdaWithError11()
+        {
+            var source =
+@"using System.Linq;
+
+public static class Program
+{
+    public static void Main()
+    {
+        var x = new {
+            X = """".Select(c => c.
+            Y = 0,
+        };
+    }
+}
+";
+            var compilation = CreateCompilationWithMscorlibAndSystemCore(source);
```

Comment:
I prefer not to add assertions that we neither want to assert nor are the point of the test. While I agree that adjusting the test when the compiler changes is a small task and so not so bad, it is multiplied by the number of times we have burdened ourselves with such unnecessary assertions. Rather than being not so bad, I'd rather not be bad at all.

---
In reply to: [66187962](https://github.com/dotnet/roslyn/pull/11807#discussion_r66187962) [](ancestors = 66187962,66187790)